Source/ZoomNet/ZoomOAuthClient.cs

```diff
@@ -0,0 +1,132 @@
+using Pathoschild.Http.Client;
+using Pathoschild.Http.Client.Extensibility;
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using ZoomNet.Json;
+using ZoomNet.Models;
+using ZoomNet.Utilities;
+
+namespace ZoomNet
+{
+	/// <summary>
+	/// REST client for interacting with Zoom's OAUTH endpoints.
+	/// </summary>
+	public class ZoomOAuthClient : IZoomOAuthClient, IDisposable
+	{
+		private const string ZOOM_OAUTH_BASE_URI = "https://zoom.us/oauth";
+
+		private HttpClient _httpClient;
+		private Pathoschild.Http.Client.IClient _fluentClient;
+
+		#region CTOR
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ZoomOAuthClient"/> class.
+		/// </summary>
+		/// <param name="connectionInfo">Connection information.</param>
+		/// <exception cref="ArgumentNullException">ArgumentNullException.</exception>
+		/// <exception cref="ZoomException">ZoomException.</exception>
+		public ZoomOAuthClient(IConnectionInfo connectionInfo)
+		{
+			if (connectionInfo == null) throw new ArgumentNullException(nameof(connectionInfo));
+
+			_httpClient = new HttpClient();
+
+			_fluentClient = new FluentClient(new Uri(ZOOM_OAUTH_BASE_URI), _httpClient);
+
+			_fluentClient.Filters.Remove<DefaultErrorFilter>();
+
+			// Remove all the built-in formatters and replace them with our custom JSON formatter
+			_fluentClient.Formatters.Clear();
+			_fluentClient.Formatters.Add(new JsonFormatter());
+
+			if (connectionInfo is OAuthTokenRevocationConnectionInfo oAuthTokenRevocationConnectionInfo)
+			{
+				var tokenHandler = new OAuthTokenHandlerForRevocation(oAuthTokenRevocationConnectionInfo, _httpClient);
+				_fluentClient.Filters.Add(tokenHandler);
+				_fluentClient.SetRequestCoordinator(new ZoomRetryCoordinator(new Http429RetryStrategy(), tokenHandler));
+			}
+			else
+			{
+				throw new ZoomException($"{connectionInfo.GetType()} is an unknown connection type", null, null, null, null);
+			}
+
+			_fluentClient.Filters.Add(new ZoomErrorHandler());
+		}
+
+		/// <summary>
+		/// Finalizes an instance of the <see cref="ZoomOAuthClient"/> class.
+		/// </summary>
+		~ZoomOAuthClient()
+		{
+			// The object went out of scope and finalized is called.
+			// Call 'Dispose' to release unmanaged resources
+			// Managed resources will be released when GC runs the next time.
+			Dispose(false);
+		}
+
+		#endregion
+
+		#region PUBLIC METHODS
+
+		/// <summary>
+		/// Revokes the access token of the user.
+		/// </summary>
+		/// <param name="cancellationToken">CancellationToken.</param>
+		/// <returns>TokenRevocation object.</returns>
+		public Task<TokenRevocation> RevokeAsync(CancellationToken cancellationToken)
+		{
+			return _fluentClient
+				.PostAsync("https://zoom.us/oauth/revoke")
```

Comment:
it looks like we don't need to provide `https://zoom.us/oauth/` here, right?

Reply:
That is correct and has been removed